### DISCUSSION TITLE
V2.0 deployment costs on Mainnet

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#540] [General]
## Costs

I used this [smart contract deployment cost estimator](https://www.rareskills.io/smart-contract-deployment-cost-calculator) to estimate the gas costs for deploying our contracts to Ethereum Mainnet. These are based on a [max gas unit price](https://www.blocknative.com/gas-estimator) of 20 gwei.

| Name | USD Cost |
|--------|--------|
| `Comptroller` | $16 |
| `LockupDynamic` | $200 |
| `LockupLinear` | $158 |
| `NFTDescriptor` | $197 | 
| Total | $571 | 

Not cheap, but not super expensive either.

## Data

- [Build sizes](https://app.warp.dev/block/MXLMdBFMvjfHdSI2m1839x)
- [Screenshot](https://github.com/sablier-labs/v2-core/assets/8782666/aa7f112d-fd2b-4c12-bafc-5f29d098d51f)

**Comment by PaulRBerg** [comment]
Here are the costs for the first mainnet deployment, which was made with a max fee per gas of 16.25 gwei, and a time when the price of ETH was ~$1,955:

| Contract | ETH Cost | Dollar Cost |
|--------|--------|--------|
| Comptroller | 0.0048587 ETH | $9.50 |
| LockupDynamic | 0.08244 ETH | $161.24 |
| LockupLinear | 0.06491618 ETH | $126.97 |
| NFTDescriptor | 0.08493285 ETH | $166.12 |
| Total | 0.23714773 ETH | $463.84 |

And here are the deployment txs:

- https://etherscan.io/tx/0x3fb9df48c76fb59e7ebc6d6aa7ce0589c1d9769043ceccc326274fa6ded986b6
- https://etherscan.io/tx/0xcb15b54cfb8f6ab2c3f9181e814301520b5da4775d1190a8587efb104c78c477
- https://etherscan.io/tx/0xd4c574cff52cde6661337beb45d2389009199a55df2e301c7301da10747f4f44
- https://etherscan.io/tx/0x14964e6a753c1dc3a65dac2aaa92dcfd08606d66299b022225ffb5ff11f2099f

**Comment by smol-ninja** [comment]
Since we had seed round, you can now stop worrying about the deployment cost 😉

**Reply by PaulRBerg** [reply]
haha yes; still informative to know the costs, e.g. for third-party deployments
